### PR TITLE
decompile brightness.c

### PR DESCRIPTION
**Pull request by red031000**

### CODE REVIEW

File: asm/include/overlay_07.inc [hunk 1]
```diff
@@ -41,10 +41,10 @@
 .public sub_02009414
 .public sub_020094BC
 .public G2dRenderer_SetSubSurfaceCoords
-.public sub_0200B484
-.public sub_0200B528
-.public sub_0200B594
-.public sub_0200B5C0
+.public StartBrightnessTransition
+.public InitAllScreenBrightnessData
```
Comment (adrienntindall) [inline asm/include/overlay_07.inc:45]: ScreenBrigtness(Data)_Init(All)

File: include/brightness.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#ifndef POKEHEARTGOLD_BRIGHTNESS_H
+#define POKEHEARTGOLD_BRIGHTNESS_H
+
+#include "global.h"
+#include <nitro/gx/g2.h>
+
+#define SCREEN_MASK_MAIN 1
+#define SCREEN_MASK_SUB 2
```
Comment (adrienntindall) [inline include/brightness.h:8]: Spacing

File: include/brightness.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    s8 transitionDirection;
+    s16 brightnessDiff;
+    s16 stepSizeInteger;
+    u16 stepSizeFractional;
+    u16 fractionalCount;
+    fx16 currentBrightness;
+    BOOL transitionActive;
+} BrightnessData;
+
+void DoBrightnessTransitionStep(BrightnessData *brightness);
```
Comment (adrienntindall) [inline include/brightness.h:25]: We could have a Brightness_ prefix for all these functions, not really required but IIRC that's what the style is supposed to be

File: src/brightness.c [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+#include "brightness.h"
+#include <nitro/gx/g2.h>
+
+static BrightnessData sMainScreenBrightnessData;
+static BrightnessData sSubScreenBrightnessData;
+
+void DoBrightnessTransitionStep(BrightnessData *brightness) {
+    BOOL transitionFinished = FALSE;
+
+    if (brightness->targetBrightness != brightness->currentBrightness + brightness->transitionDirection * brightness->stepSizeInteger && brightness->currentBrightness != brightness->targetBrightness) {
```
Comment (adrienntindall) [inline src/brightness.c:11]: First conditional should use parenthesis since it's large

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: include/brightness.h (hunk 3 above)
Comment (red031000) [reply include/brightness.h:25]: I'd rather not tbh, since it's pretty clear what these funcs belong to

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**